Skuid_Techniques/Navigation/README.MD

```diff
@@ -0,0 +1,167 @@
+
+Here is how to handle the requirements listed above. 
+
+### Highlight the users current selection  
+First, configure the design system so the selected state actually is different.  Look for Selected State under the Navigation element node of your design system.  Change the background color, or text font… For your testing make it obvious. 
+
+The navigation components provide a standard selected state support.  But you can enhance this by using display logic (look for Selected conditions).  Maybe you want multiple entries selected, or you want to drive selection from another component - like a wizard button.   
+
+Note that selecting a second nav element will clear all Selected state conditions regardless of whether the selected state condition still is true.  (This is being reported as an enhancement)  
+
+### Display data in the navigation element 
+You can include model data in the labels of your navigation component.  Let users know how many open tasks there are before they open that navigation element. How many questions are still outstanding in that section of the survey. You can also use conditional merge syntax to show or hide HTML elements in the navigation labels based on underlying data.  Show that goofy smiley face icon if all tasks are complete.  
+
+Note that the navigation component is not connected to a model,  so you will have to use global merge syntax  `{{$Model.modelname.data.0.fieldname}}`.  In V2, components do not yet listen to changes of data through global merge syntax.  The indicators will not change unless you reload the page.  (Coming enhancement) 
+
+This challenge might point you to using the data iterator components for your navigation. See below. 
+
+### Disable navigation element based on data.  
+We do not have Enable Condition logic for navigation items.  (Enhancement!) Instead use Branch Actions to disable the navigation item by not allowing the primary actions to be completed. You have to define the branch logic for each navigation item,  but this allows you to have different logic rules for each item.  The failure branch can post a message about why the item is disabled.  
+
+### Collapsible navigation 
+You will need to create two copies of the component (one with label and the other without) and use a ui-only field to control which shows.  Put a form control at the bottom of your navigation to let the user indicate if they want the collapsed or expanded navigation. 
+
+When global model listening is implemented - use conditional merge syntax to show or hide the label.
+```xml
+{{#$Model.UIModel.data.0.CollapsedNav}}””{{/$Model.UIModel.data.0.CollapsedNav}}{{^$Model.UIModel.data.0.CollapsedNav}}Label Title{{/$Model.UIModel.data.0.CollapsedNav}}
+```
```

Comment:
I love this kind of merge syntax wizardry!

Reply:
Thanks.